components/molecules/TranslationsWrapper.tsx

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { SupportedLocale } from '../../@types';
+
+type Props = {
+  locale: SupportedLocale
+  children: JSX.Element
+};
+
+export const LocaleContext = React.createContext<{ locale: SupportedLocale }>({ locale: 'en' });
+export const TranslationsWrapper = ({ locale, children }: Props) => {  
+  return (
+    <LocaleContext.Provider value={ { locale } } >
+      { children } 
+    </LocaleContext.Provider>
```

Comment:
Note to ask later: Is there a reason you defined the provider as a molecule and wrapped each component in it, instead of wrapping the whole app (root component) with this provider?